File: template/elixir/test/aoc_test.exs
defmodule AocTest do
  use ExUnit.Case
  doctest Aoc

  test "solves part one" do
    assert Aoc.solve_part_one("") == 0
  end

  test "solves part two" do
    assert Aoc.solve_part_two("") == 0
  end
end

File: template/scala/test/aoc.test.scala
class AocSuite extends munit.FunSuite:
  /** Checks one solution function against the example in the puzzle statement. */
  private def checkSolution(solve: String => Long, input: String, expected: Long): Unit =
    assertEquals(solve(input), expected, s"failed on input $input")

  test("part one solves the example"):
    checkSolution(solvePartOne, "", 0L)

  test("part two solves the example"):
    checkSolution(solvePartTwo, "", 0L)
